qcodes/station.py

```diff
@@ -485,7 +485,14 @@ def setup_parameter_from_dict(instr: Instrument, name: str,
                     setattr(parameter, attr, val)
                 # extra attributes that need parsing
                 elif attr == 'limits':
-                    lower, upper = [float(x) for x in val.split(',')]
+                    if isinstance(val, str):
+                        issue_deprecation_warning(
+                            ('use of a comma separated string for the limits'
+                             'keyword'),
+                            alternative='an array like "[lower_lim, upper_lim]"')
+                        lower, upper = [float(x) for x in val.split(',')]
+                    else:
+                        lower, upper = val
```

Comment:
I was about to ask "shall we add some validation here so that if a user incorrectly supplied the limits (for example, only one limit is given), then the error message is more clear than 'could not unpack a list of 1 element into two variables'" - but then i realized that a schema is coming, hence all the validation will be done via the schema. Is my understanding correct?

Reply:
Exactly :-)